teams/vsp/teams/tools/frontend/cypress-migration-guide.md

```diff
@@ -0,0 +1,258 @@
+
+For additional ways to interact with the DOM, we have also included the [Cypress Testing Library](https://testing-library.com/docs/cypress-testing-library/intro) as a dependency.
+
+### Custom Commands <a name="custom-commands"></a>
+
+Some custom Nightwatch commands located in `src/platform/testing/e2e/nightwatch-commands` were converted to custom Cypress commands, and can be found in `src/platform/testing/e2e/cypress/support/commands`.
+
+Many of them did not need to be converted, either because Cypress supports their functionality natively or the functionality was [too simple to abstract into a command](https://docs.cypress.io/api/cypress-api/custom-commands.html#2-Don%E2%80%99t-overcomplicate-things).
+
+These custom commands can be called from the `cy` object. For example:
+
+```javascript
+cy.axeCheck();
+```
+
+### Test Data (Fixtures) <a name="test-data"></a>
+
+In Cypress, because everything in a test is executed inside of the browser, [fixtures](https://docs.cypress.io/api/commands/fixture.html#Syntax) are used to get access to data in tests.
+
+By default, Cypress doesn't support fixtures in separate directories, so we have a custom command for accessing fixtures stored in your app's directory.
+
+For example, `src/applications/hca/tests/schema` contains test data for the `hca` application. You can load the file as a fixture like so:
+
+```javascript
+cy.syncFixtures({
+  'data': 'srce/applications/hca/tests/schema',
+  'minimal-test.json': 'src/applications/hca/tests/schema/maximal-test.json', 
+  'maximal-test': 'src/applications/hca/tests/schema/maximal-test.json',
+});
+
+cy.fixture('maximal-test.json').as('testData');
+```
+
+To access the contents of the file, you can use a combination of `cy.get()` and `cy.then()`:
+
+```javascript
+cy.get('@testData').then(testData => {
+  cy.get('#root_firstName').type(testData.veteranFullName.first);
+});
+```
+
+### Mocking Data <a name="mocking-data"></a>
+
+Cypress allows you to [stub responses](https://docs.cypress.io/guides/guides/network-requests.html#Stubbing), avoiding the need for mock API helpers.
+
+```javascript
+// Start a server to begin routing responses
+cy.server();
+
+// Stub request to get enrollment status
+cy.route({
+  method: 'GET',
+  url: '/v0/health_care_applications/enrollment_status*',
+  status: 404,
+  response: {
+    errors: [
+      {
+        title: 'Record not found',
+        detail: 'The record identified by  could not be found',
+        code: '404',
+        status: '404',
+      },
+    ],
+  },
+}).as('getApplication');
+
+// Stub request to submit application
+cy.route('POST', '/v0/health_care_applications', {
+  formSubmissionId: '123fake-submission-id-567',
+  timestamp: '2016-05-16',
+}).as('submitApplication');
+
+// Workaround to intercept requests made with fetch API.
+cy.on('window:before:load', window => {
+  delete window.fetch;
+});
```

Comment:
```suggestion
```

Devs won't need to add this to tests, since it's being done within the support file.